More Code Examples from the Programming Phoenix book

nelsonic opened the pull request:
@iteles adding code examples for the Programming Phoenix book 
(_I've personally written and tested the examples and been opening issues along the way e.g #38 so other people can learn **faster**..._) 👍 

Fixes https://github.com/dwyl/learn-phoenix-framework/issues/35, https://github.com/dwyl/learn-phoenix-framework/issues/49 & https://github.com/dwyl/learn-phoenix-framework/issues/50

Please review/merge when you can thanks.

shouston3 commented on rumbl/priv/repo/seeds.exs:
```diff
@@ -9,3 +9,11 @@
+
+for category <- ~w(Action Drama Romance Comedy Sci-fi) do
+  Repo.insert!(%Category{name: category}) ||
+    Repo.insert!(%Category{name: category})
```
I think these should be different, I have:

```
  Repo.get_by(Category, name: category) ||
    Repo.insert!(%Category{name: category})
```

instead

nelsonic replied:
@shouston3 good shout. please feel free to edit the file in PR. 👍 
`else` I will fix tomorrow morning. 😉

shouston3 commented on rumbl/priv/repo/migrations/20170218093549_create_category.exs:
```diff
@@ -0,0 +1,13 @@
+    create table(:categories) do
+      add :name, :string, null: false
+
+      timestamps() # function call parenthesis not in book!
```
Nice 👍

shouston3 commented on rumbl/test/controllers/auth_test.exs:
```diff
@@ -0,0 +1,83 @@
+  end
+
+  test "login with password missmatch", %{conn: conn} do
+    _ = insert_user(username: "me", password: "secrete")
```
"secrete" typo

nelsonic replied:
yeah, worth fixing. thanks. 👍

shouston3 commented on rumbl/test/models/user_test.exs:
```diff
@@ -0,0 +1,42 @@
+  # test "changeset with invalid attributes" do
+  #   changeset = User.changeset(%User{}, @invalid_attrs)
+  #   refute changeset.valid?
+  # end
```
I also have these comments in mine

shouston3 commented on rumbl/web/controllers/video_controller.ex:
```diff
@@ -3,19 +3,28 @@ defmodule Rumbl.VideoController do
-
+  def create(conn, %{"video" => video_params}, user) do
+    IO.inspect video_params
+    IO.inspect user
```
Perhaps remove these `IO.inspects`

shouston3 commented on rumbl/web/controllers/video_controller.ex:
```diff
@@ -3,19 +3,28 @@ defmodule Rumbl.VideoController do
+    user
+      |> build_assoc(:videos)
+      |> Video.changeset(video_params)
+    IO.inspect changeset
```
`IO.inspect`

nelsonic replied:
Will remove. in next commit. 👍

shouston3 commented on rumbl/web/models/video.ex:
```diff
@@ -6,16 +6,23 @@ defmodule Rumbl.Video do
+    belongs_to :category, Rumbl.Category
 
-    timestamps()
+    timestamps() # invocation parenthesis not in book
```
👍

des-des replied:
Ugh this has been confusing us with warnings

nelsonic replied:
@des-des are you running the latest version of Phoenix?

des-des replied:
@nelsonic yeah 1.2.1

des-des commented on rumbl/web/static/js/player.js:
```diff
@@ -0,0 +1,29 @@
+let Player = {
```
JS -> Elm?!

nelsonic replied:
@des-des this example is from "The Book" ...
I don't _intend_ to keep the ES6 after I've finished writing out the code. 😉

des-des replied:
hehe

nelsonic commented:
Totes works!
![image](https://cloud.githubusercontent.com/assets/194400/23590950/c3dd01ba-01e0-11e7-91e9-d06165bc484e.png)

nelsonic commented:
@iteles I've addressed the review comments from @shouston3  & @des-des 
and added Travis-CI to confirm that the demo app is working.
Please take a look at the code when you have time and merge if ok. 
thanks! 👍

des-des commented on rumbl/web/static/js/player.js:
```diff
@@ -1,6 +1,7 @@
-let Player = {
-  player: null,
-
+let Player = {  // JS works for small scripts like this
```
:heart:

iteles replied:
Also loving this pragmatism 👍
